load-test-framework/src/main/java/com/google/pubsub/flic/Driver.java

```diff
@@ -208,29 +239,122 @@ private void run() {
       synchronized (pollingExecutor) {
         pollingExecutor.shutdownNow();
       }
-      printStats(gceController.getStatsForAllClientTypes());
-
+      
+      List<String[]> stats = getStatsFromResults(gceController.getStatsForAllClientTypes());
+      printStats(stats);
+      
+      if (SPREADSHEET_ID.length() < 0) {
+        // Output results to common Google sheet
+        Sheets sheetService = getSheetsService();
+        sendToSheets(sheetService, stats);
+      }
       gceController.shutdown(null);
       System.exit(0);
     } catch (Throwable t) {
       log.error("An error occurred...", t);
       System.exit(1);
     }
   }
-
-  private void printStats(Map<ClientType, Controller.LoadtestStats> results) {
+  
+  private List<String[]> getStatsFromResults(Map<ClientType, Controller.LoadtestStats> results) {
+    List<String[]> out = new ArrayList<String[]>(results.size());
     results.forEach((type, stats) -> {
-      log.info("Results for " + type + ":");
-      log.info("50%: " + LatencyDistribution.getNthPercentile(stats.bucketValues, 50.0));
-      log.info("99%: " + LatencyDistribution.getNthPercentile(stats.bucketValues, 99.0));
-      log.info("99.9%: " + LatencyDistribution.getNthPercentile(stats.bucketValues, 99.9));
-      // CPS Publishers report latency per batch message.
-      log.info("Average throughput: " +
-          new DecimalFormat("#.##").format(
-              (double) LongStream.of(
-                  stats.bucketValues).sum() / stats.runningSeconds * messageSize / 1000000.0
-                  * (type.isCpsPublisher() ? cpsPublishBatchSize : 1)) + " MB/s");
+      String[] row = new String[5];
+      row[0] = type.toString();
+      row[1] = LatencyDistribution.getNthPercentile(stats.bucketValues, 50.0);
+      row[2] = LatencyDistribution.getNthPercentile(stats.bucketValues, 99.0);
+      row[3] = LatencyDistribution.getNthPercentile(stats.bucketValues, 99.9);
+      row[4] = new DecimalFormat("#.##").format(
+          (double) LongStream.of(
+              stats.bucketValues).sum() / stats.runningSeconds * messageSize / 1000000.0
+              * (type.isCpsPublisher() ? cpsPublishBatchSize : 1));
     });
+    return out;
+  }
+
+  private void printStats(List<String[]> stats) {
+    for(String[] row : stats) {
+      log.info("Results for " + row[0] + ":");
+      log.info("50%: " + row[1]);
+      log.info("99%: " + row[2]);
+      log.info("99.9%: " + row[3]);
+      // CPS Publishers report latency per batch message.
+      log.info("Average throughput: " + row[4] + " MB/s");
+    }
+  }
+  
+  private Sheets getSheetsService() throws Exception {
```

Comment:
This belongs in a separate file. Why don't you make a com.google.pubsub.flic.output.Sheets and put everything in there, and have a public function that takes the List<String[]> stats and writes it out, this is too much for Driver